sdk/python/kfp/v2/compiler/compiler_utils.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""KFP v2 DSL compiler utility functions."""
+
+from typing import List, Mapping
+from kfp.v2 import dsl
+from kfp.v2.proto import pipeline_spec_pb2
+
+
+def build_runtime_parameter_spec(
+    pipeline_params: List[dsl.PipelineParam]
+) -> Mapping[str, pipeline_spec_pb2.PipelineSpec.RuntimeParameter]:
+  """Converts pipeine parameters to runtime parameters mapping.
+
+  Args:
+    pipeline_params: The list of pipeline parameters.
+
+  Returns:
+    A map of pipeline parameter name to its runtime parameter message.
+  """
+  def to_message(param: dsl.PipelineParam):
+    result = pipeline_spec_pb2.PipelineSpec.RuntimeParameter()
+    if param.param_type == 'Integer' or (param.param_type is None and
```

Comment:
Is it possible that `param.param_type` and the type of `param.value` are inconsistent? Do we have a check somewhere else?

Reply:
Yes, it's entirely possible, and we don't have any check. 
Be aware that `param.param_type` gets its value from Python type annotation. I think this, if presents, should be the user intend, thus giving it a higher ranked vote.
The value of a runtime parameter may also be provided later, outside the SDK.There's always a possibility of type mismatching and we can do little about it.

Reply:
I'm okay if we leave this as-is. So treat this comment as non-blocking.

> The value of a runtime parameter may also be provided later, outside the SDK.There's always a possibility of type mismatching and we can do little about it.

Here the value is actually the default value, right? For some reason `dsl.PipelineParam` does not distinguish this two cases but I think in the v2 use case most likely it is. If this is the default value, which is also inferred from the pipeline function signature, it should be in consistent with the declared type, right?

Reply:
That's fair. I'll follow up on this later. Thanks!